examples/step-27/doc/results.dox

```diff
@@ -209,7 +209,71 @@ include estimating the smoothness not on single cells, but cell assemblies or
-further research on finding good $hp$ refinement criteria. On the other hand,
-the main point of the current program was to demonstrate using the $hp$
-technology in deal.II, which is unaffected by our use of a possible
+further research on finding good $hp$-refinement criteria. On the other hand,
+the main point of the current program was to demonstrate using the
+$hp$-technology in deal.II, which is unaffected by our use of a possible
 sub-optimal refinement criterion.
+
+
+
+<a name="extensions"></a>
+<h3>Possibilities for extensions</h3>
+
+<h4>Different hp-decision strategies</h4>
+
+This tutorial demonstrates only one particular strategy to decide between h- and
+p-adaptation. In fact, there are many more ways to automatically decide on the
+adaptation type from which a couple are already implemented in deal.II:
+<ul>
+  <li><i>Fourier coefficient decay:</i> This is the strategy currently
+  implemented in this tutorial. For more information on this strategy, see
+  the general documentation of the SmoothnessEstimator::Fourier namespace.</li>
+
+  <li><i>Legendre coefficient decay:</i> This strategy is quite similar
+  to the current one, but uses Legendre series expansion rather than the
+  Fourier one: instead of sinusoids as basis functions, this strategy uses
+  Legendre polynomials. The transition from the Fourier strategy to
+  the Legendre one is quite simple: You just need to change the series
+  expansion class and the corresponding smoothness estimation function
+  to be part of the proper namespaces FESeries::Legendre and
+  SmoothnessEstimator::Legendre. For the theoretical background of this
+  strategy, consult the general documentation of the
+  SmoothnessEstimator::Legendre namespace, as well as @cite mavriplis1994hp ,
+  @cite eibner2007hp and @cite davydov2017hp.</li>
+
+  <li><i>Refinement history:</i> The last strategy is quite different
+  from the other two. In theory, we know how the error will converge
+  after changing the discretization of the function space. With
+  h-refinement, the solution converges algebraically as already pointed
+  out in step-7. It the solution is sufficiently smooth, though, we
+  expect that the solution will converge exponentially with increasing
+  polynomial degree of the finite element. We can compare a proper
+  prediction of the error with the actual error in the following step to
+  see verify if our choice of adaptation type was justified. The transition
+  to this strategy is a bit more complicated. For this, we need an
+  initialization step with pure h- or p-refinement and we need to transfer
+  the predicted errors over adapted meshes. The extensive documentation
+  of the hp::Refinement::predict_error() function describes not only the
+  theoretical details of this approach, but also presents a blueprint on
+  how to implement this strategy in your code. Note that this strategy
+  is only suited for time-independent problems or to create an initial
```

Comment:
I don't think I understand this comment. I *think* that what you're trying to say is that one can't predict the smoothness of the solution at the next time step, but if one wanted to build the mesh for just one time step (e.g., start each step with a coarse mesh, then keep refining until we're happy with the solution, and only then moving on to the next time step), then this approach here could be used as well.

I think that your comment is not actually specific to this strategy but all the others as well.

Reply:
You are right, this short comment does not give the issue justice that I wanted to address. I've extended the documentation.